File: examples/thymeleaf-java-example/src/main/java/de/tschuehly/example/thymeleafjava/web/simple/SimpleViewComponent.java
package de.tschuehly.example.thymeleafjava.web.simple;

import de.tschuehly.example.thymeleafjava.core.ExampleService;
import de.tschuehly.spring.viewcomponent.core.component.ViewComponent;
import de.tschuehly.spring.viewcomponent.thymeleaf.ViewContext;

@ViewComponent
public class SimpleViewComponent {
    final ExampleService exampleService;

    public SimpleViewComponent(ExampleService exampleService) {
        this.exampleService = exampleService;
    }

    public record SimpleView(String helloWorld) implements ViewContext {
    }

    public SimpleView render() {
        return new SimpleView(exampleService.getHelloWorld());
    }
}

File: examples/thymeleaf-kotlin-example/src/main/kotlin/de/tschuehly/example/thymeleafkotlin/web/simple/SimpleViewComponent.kt
package de.tschuehly.example.thymeleafkotlin.web.simple

import de.tschuehly.example.thymeleafkotlin.core.ExampleService
import de.tschuehly.spring.viewcomponent.core.component.ViewComponent
import de.tschuehly.spring.viewcomponent.thymeleaf.ViewContext

@ViewComponent
class SimpleViewComponent(
    val exampleService: ExampleService
) {
    fun render() = SimpleView(exampleService.getHelloWorld())

    data class SimpleView(val helloWorld: String) : ViewContext

}